pkg/cache/v3/simple.go

```diff
@@ -26,6 +26,35 @@ import (
 	"github.com/envoyproxy/go-control-plane/pkg/server/stream/v3"
 )
 
+// ResourceSnapshot is an abstract snapshot of a collection of resources that
+// can be stored in a SnapshotCache. This enables applications to use the
+// SnapshotCache watch machinery with their own resource types. Most
+// applications will use Snapshot.
+type ResourceSnapshot interface {
```

Comment:
If this is an interface at this high of a level why not just call it a `Snapshot`? I think that would make more sense and then subsequent structs that implement can name themselves accordingly. I imagine the snapshot itself as the blueprint for a grouping of resources looks like.

Changing this would require we change the original snapshot struct though which I'm not opposed to

Reply:
Yeh I didn't want to rename `Snapshot`, because it's pretty disruptive and I also didn't have a good suggestion for a new name :)

Reply:
Hm, what are your thoughts on if we were to deal with that upfront cost? You think the API would be easier to use/consume?

Reply:
I'm not sure that there are big wins to the API from renaming. It's probably a bit nicer, but not sure it's really a big deal. If we have a a larger goal to separate out the envoy protobuf dependencies, maybe we would think of it as part of that.

Reply:
@alecholmez I'm  still inclined to keep the current `Snapshot` name as is for now.